docs/design/features/readytorun-elf-support-roadmap.md

```diff
@@ -0,0 +1,170 @@
+Roadmap to native Linux executable format (ELF) support in Ready to Run
+=======================================================================
+
+Revisions:
+* 1.1 - [Tomas Rylek](https://github.com/trylek) - 2/27/2020
+
+# Introduction
+
+This document details the steps needed for supporting native Linux executable
+format (ELF) for ready-to-run binaries on Linux. Today, ready-to-run binaries
+use the Windows PE executable format on all three supported platforms
+(Windows, Linux, OSX). Due to this fact ready-to-run binaries are "unnatural"
+on non-Windows platforms incurring both performance and user experience costs
+(incomplete integration with OS-native application management, inability to
+use OS executable loader, lack of OS-native diagnostic experience). This
+proposal tries to summarize the steps needed towards native executable support.
+
+# Scope of changes
+
+Support for Linux native executable format will require changes to the following
+areas:
+
+* **CoreCLR runtime**: ideally we should be able to use the Linux ELF headers
+  in the native code. The current treatment of ready-to-run files involving
+  classes PEFile, PEImage and PEImageLayout aka PEDecoder, need virtualizing
+  to allow for their ELF counterparts and proper deduplication of common logic
+  especially related to R2R that should be mostly autonomous once the R2R header
+  has been located. Fully switching over to native ELF format for ready-to-run
+  binaries is more complicated as it involves not yet implemented embedded MSIL
+  in native R2R binaries, a new concept we're currently developing as part of
+  the composite R2R effort; this will replace the existing lookup for COR header
+  using the COM descriptor directory entry in the PE header.
+
+* **Crossgen2 compiler**: The SectionBuilder and R2RPEBuilder needs to be
+  decoupled from the PE format by some form of virtualization. We need to
+  port the various ELF header structures to managed code for use in Crossgen2
+  and R2RDump (do we need some approval or special arrangements for porting
+  the Linux header files like elf.h and elftypes.h to managed code in the runtime
+  repo?). After we have the constants, structures and their serialization
+  mechanisms in place and after we figure out the proper abstraction the work
+  should be pretty straightforward in terms of just filling in the various header
+  and section tables and emitting the section data in the proper place.
+
+* **Diagnostics / SOS**: switching over to native ELF will have fundamental effect on
```

Comment:
I do not see why switching to ELF envelope for R2R has fundamental effect on SOS

Reply:
OK, JanV also mentioned offline some of these might be no-concerns.